### PR TITLE
Modularize libproc macro

### PR DESCRIPTION
**Pull request by P-E-P**
Refactor a bit the global structure of libproc_macro to make is more usable and fix the link issue against the compiler.

### CODE REVIEW

**Review by philberty — COMMENTED** [review]
I think ProcMacro camel case would be better but I will leave it up to you as you are the main author :dancers:

**Comment by P-E-P** [comment]
> I think ProcMacro camel case would be better but I will leave it up to you as you are the main author dancers

I wasn't too sure, but you got me decided :)

File: gcc/rust/Make-lang.in [hunk 1]
```diff
@@ -355,7 +358,8 @@ RUST_INCLUDES = -I $(srcdir)/rust \
 	-I $(srcdir)/rust/checks/errors \
 	-I $(srcdir)/rust/checks/errors/privacy \
 	-I $(srcdir)/rust/util \
-        -I $(srcdir)/rust/metadata
+        -I $(srcdir)/rust/metadata \
+		-I $(srcdir)/../libgrust/libproc_macro
```
Comment (CohenArthur) [inline gcc/rust/Make-lang.in:362]: I wonder if we want `../libgrust/libproc_macro` here or just `../libgrust`: My reasoning is, if we go for `-I $(srcdir)/../libgrust`, we won't have to touch this line again even if we create more libraries that both compiler and users will want to use, and it also makes it clearer at the include site that we're including "proc macro related" headers:

```c++
#include "libprocmacro/ident.h"
// vs
#include "ident.h"
```

**Review by CohenArthur — APPROVED** [review]
The code changes look good to me. The build system changes look good too but I'm less able to review them meaningfully. Great work :)

File: gcc/rust/Make-lang.in (hunk 1 above)
Comment (P-E-P) [reply gcc/rust/Make-lang.in]: Very well, I changed it.

**Review by CohenArthur — APPROVED** [review]
LGTM!

**Pull request merged**